src/openswathalgo/include/OpenMS/OPENSWATHALGO/ALGO/StatsHelpers.h

```diff
@@ -43,14 +43,22 @@ namespace OpenSwath
   @brief compute dotprod of vectors
   */
   template <typename Texp, typename Ttheo>
-  double dotProd(Texp intExpBeg, Texp intExpEnd, Ttheo intTheo)
-  {
-    std::vector<double> res(std::distance(intExpBeg, intExpEnd));
-    std::transform(intExpBeg, intExpEnd, intTheo, res.begin(), std::multiplies<double>());
-    double sum = std::accumulate(res.begin(), res.end(), 0.);
```

Comment:
So, was one of those two the culprit?

Reply:
Yes or their combination. It doesn't look like there was UB anywhere but maybe too much rounding going on with all the intermediate doubles this creates.